services/121-service/src/payments/reconciliation/onafriq-reconciliation/onafriq-reconciliation.service.ts

```diff
@@ -102,4 +118,103 @@ export class OnafriqReconciliationService {
     }
     return OnafriqTransactionStatus.other;
   }
+
+  public async sendReconciliationReport(): Promise<number> {
+    const isTest = IS_DEVELOPMENT; // Use this work-around to make cronjob.test.ts work
+    const result =
+      await this.generateAndSendReconciliationReportYesterday(isTest);
+    return result.length;
+  }
+
+  public async generateAndSendReconciliationReportYesterday(
+    isTest?: boolean,
+  ): Promise<OnafriqReconciliationReport[]> {
+    let where = {};
+    if (!isTest) {
+      // ##TODO: should we account for how to handel this in acceptance test? As it will require us to wait a day now.
+      const yesterdayStart = new Date();
+      yesterdayStart.setUTCDate(yesterdayStart.getUTCDate() - 1); // In production use yesterday
+      yesterdayStart.setUTCHours(0, 0, 0, 0);
+      const yesterdayEnd = new Date(yesterdayStart);
+      yesterdayEnd.setUTCHours(23, 59, 59, 999);
+      where = {
+        transaction: {
+          created: Between(yesterdayStart, yesterdayEnd),
+        },
+      };
+    }
+
+    const onafriqTransactions =
+      await this.onafriqTransactionScopedRepository.find({
+        where,
+        relations: ['transaction'],
+      });
+
+    const report: OnafriqReconciliationReport[] = onafriqTransactions.map(
+      (onafriqTransaction) =>
+        OnafriqReconciliationMapper.mapTransactionToReportItem(
+          onafriqTransaction,
+        ),
+    );
+
+    // ##TODO: is it fine to not do this at all in test?
```

Comment:
Maybe the easiest thing would be to test this with a unit test? Or another option would be to have line 176 give the csv return instead of a JSON and rewite the API test for that

Reply:
agreed to leave it be for now